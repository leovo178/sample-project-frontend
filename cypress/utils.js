/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
