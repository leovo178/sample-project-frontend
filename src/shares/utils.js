const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
