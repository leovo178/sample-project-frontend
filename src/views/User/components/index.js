export { default as WelcomeDialog } from './WelcomeDialog'

export { default as UserList } from './UserList'

export { default as UserFormEditor } from './UserFormEditor'

export { default as SearchUserBox } from './SearchUserBox'
