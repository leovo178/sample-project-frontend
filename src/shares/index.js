import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
