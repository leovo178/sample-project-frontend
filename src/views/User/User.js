import React, { useState } from 'react'

import { Box, Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

import { WelcomeDialog, UserList, UserFormEditor } from './components'

const useStyles = makeStyles(theme => ({
	root: {
		width: '100%',
		height: '100vh',
	},
	container: {
		padding: theme.spacing(3),
		height: '100vh',
	},
	item__signup: {
		display: 'flex',
		alignItems: 'center',
		height: '100%',
		border: `1px solid ${theme.palette.common.border}`,
		marginRight: theme.spacing(1.5),
		position: 'relative',
	},
}))

const User = () => {
	const classes = useStyles()

	const [dialogVisible, setDialogVisible] = useState(true)

	return (
		<Box className={classes.root}>
			<Grid container className={classes.container}>
				<Grid item xs={4}>
					<Box className={classes.item__signup}>
						{dialogVisible ? (
							<WelcomeDialog onCreateUser={() => setDialogVisible(false)} />
						) : null}
						<UserFormEditor />
					</Box>
				</Grid>
				<Grid item xs={8}>
					<UserList onSelectUser={() => setDialogVisible(false)} />
				</Grid>
			</Grid>
		</Box>
	)
}

export default User
