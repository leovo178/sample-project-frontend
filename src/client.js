import { ApolloClient } from 'apollo-client'
import { createUploadLink } from 'apollo-upload-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { setContext } from 'apollo-link-context'

import { getToken } from './shares/utils'
import { APOLLO_SERVER } from './configs.local'

const typeDefs = {}

const resolvers = {
	Mutation: {
		setUserSearchValue: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValue: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUser: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUser,
				},
			})
			return selectedUser
		},
		setUploadedFile: (_, { file }, { cache }) => {
			if (file) {
				cache.writeData({
					data: {
						file,
					},
				})
			}
			return file
		},
		setUserSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValueOfMain: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUserOfMain: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUserOfMain: selectedUser,
				},
			})
			return selectedUser
		},

		setMessageSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					messageSearchValueOfMessage: searchValue,
				},
			})
			return searchValue
		},
		setMessageCreateValueOfMain: (_, { createValue }, { cache }) => {
			cache.writeData({
				data: {
					messageCreateValueOfMessage: createValue,
				},
			})
			return createValue
		},
		resetCache: (_, { data }, { cache }) => {
			cache.writeData({
				data,
			})
			return data
		},
	},
}

const httpLink = createUploadLink({
	uri: APOLLO_SERVER,
	credentials: 'same-origin',
})

const authLink = setContext((_, { headers }) => {
	const token = getToken()
	return {
		headers: {
			...headers,
			authorization: token ? `Bearer ${token}` : '',
		},
	}
})

const cache = new InMemoryCache()

const client = new ApolloClient({
	cache,
	link: authLink.concat(httpLink),
	typeDefs,
	resolvers,
})

const initialState = {
	userSearchValue: '',
	messageCreateValueOfMessage: '',
	messageSearchValueOfMessage: '',
	userSearchValueOfMain: '',
	selectedUser: {
		id: '',
		name: '',
		email: '',
		__typename: 'User',
	},
	file: {
		filename: '',
		link: '',
		__typename: 'File',
	},
	selectedUserOfMain: {
		id: '',
		name: '',
		email: '',
		__typename: 'UserOfMain',
	},
}

cache.writeData({
	data: initialState,
})

export { client, initialState }
