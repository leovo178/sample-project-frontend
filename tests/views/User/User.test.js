import React from 'react'

import { createMockClient } from 'mock-apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'

import { act, cleanup } from '@testing-library/react'

import User from '@views/User'

import { renderWithApolloClient } from '@tests/shares/utils'
import { initialState, resolvers } from '@src/client'

describe('User', () => {
	const cache = new InMemoryCache()
	cache.writeData({
		data: initialState,
	})
	const mockClient = createMockClient({
		cache,
		resolvers,
	})

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', async () => {
		jest.spyOn(window, 'alert').mockImplementation(() => 'Mock alert!!!')
		let wrapper
		await act(async () => {
			const { container } = renderWithApolloClient(<User />, mockClient)
			wrapper = container
		})

		expect(wrapper).toMatchSnapshot()
	})
})
