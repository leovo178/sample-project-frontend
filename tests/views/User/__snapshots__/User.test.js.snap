// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should match snapshot 1`] = `
<div
  style="width: 300px; height: 700px;"
>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-root-1"
  >
    <div
      class="MuiGrid-root makeStyles-container-2 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          class="MuiBox-root MuiBox-root-109 makeStyles-item__signup-3"
        >
          <div
            class="MuiBox-root MuiBox-root-113 makeStyles-root-110"
          >
            <div
              class="MuiBox-root MuiBox-root-114"
            >
              <h6
                class="MuiTypography-root makeStyles-welcometext-111 MuiTypography-subtitle1"
              >
                Welcome to Vatech !!!
              </h6>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-112 MuiButton-fullWidth"
                data-testid="create-user-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Create user
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-183 makeStyles-root-177"
          >
            <h5
              class="MuiTypography-root makeStyles-formtitle-178 MuiTypography-h5"
              data-testid="formeditor-title"
            >
              Modify
            </h5>
            <div
              class="makeStyles-formcontent-180"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-forminput-179"
                data-testid="formeditor-email-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink="true"
                >
                  EMAIL
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="true"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    placeholder="Email"
                    type="email"
                    value="steve@example.com"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-244 PrivateNotchedOutline-legendNotched-245"
                    >
                      <span>
                        EMAIL
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-forminput-179"
                data-testid="formeditor-name-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink="true"
                >
                  NAME
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="true"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    placeholder="Name"
                    type="text"
                    value="steve"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-244 PrivateNotchedOutline-legendNotched-245"
                    >
                      <span>
                        NAME
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-forminput-179"
                data-testid="formeditor-password-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  PASSWORD
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="true"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-244"
                    >
                      <span>
                        PASSWORD
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-forminput-179"
                data-testid="formeditor-password-confirm-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  PASSWORD CONFIRM
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="true"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    placeholder="Password Confirm"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-244"
                    >
                      <span>
                        PASSWORD CONFIRM
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-formbuttons-181"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-182 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
                data-testid="formeditor-submit-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Save
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-182 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
                data-testid="formeditor-delete-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Delete
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-182 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
                data-testid="formeditor-cancel-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Cancel
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          class="MuiBox-root MuiBox-root-270 makeStyles-root-265"
        >
          <div
            class="MuiBox-root MuiBox-root-271 makeStyles-userlist__container-266"
          >
            <div
              class="MuiBox-root MuiBox-root-272 makeStyles-searchbox-268"
            >
              <h5
                class="MuiTypography-root makeStyles-searchbox__title-269 MuiTypography-h5"
                data-testid="userlist-title"
              >
                User List
              </h5>
              <div
                class="MuiBox-root MuiBox-root-278 makeStyles-root-273"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-input-274 makeStyles-input-277"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      placeholder="search..."
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-243"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-275 MuiButton-containedSizeLarge MuiButton-sizeLarge"
                  data-testid="actioninputbox-button"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-icon-276"
                      data-testid="search-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-288 makeStyles-userlist__table-267"
            >
              <div
                class="MuiBox-root MuiBox-root-303 makeStyles-root-300"
                data-testid="infinitetable"
              >
                <div
                  class="MuiGrid-root makeStyles-container__header-301 MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                  >
                    <p
                      class="MuiTypography-root makeStyles-item__header-302 MuiTypography-body2"
                      data-testid="tableheader-email"
                    >
                      EMAIL
                    </p>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                  >
                    <p
                      class="MuiTypography-root makeStyles-item__header-302 MuiTypography-body2"
                      data-testid="tableheader-name"
                    >
                      NAME
                    </p>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-304 makeStyles-root-300"
                >
                  <div
                    style="overflow: visible; height: 0px; width: 0px;"
                  >
                    <div
                      class="makeStyles-overflow-305"
                      style="height: 700px; width: 200px; overflow: auto; position: relative; will-change: transform;"
                    >
                      <div
                        style="height: 11900px; width: 100%;"
                      >
                        <div
                          class="MuiGrid-root makeStyles-root-306 makeStyles-root-311 makeStyles-item__active-307 MuiGrid-container"
                          data-testid="row-5e68995fb6d0bc05829b6e79"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-312 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                steve@example.com
                              </p>
                            </div>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-313 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                steve
                              </p>
                            </div>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root makeStyles-root-306 makeStyles-root-314 MuiGrid-container"
                          data-testid="row-5e68b5c3dd814dd832cda79d"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-315 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                nvdai2355s56se@gmail.com
                              </p>
                            </div>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-316 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                asd
                              </p>
                            </div>
                          </div>
                        </div>
                        <div
                          class="MuiGrid-root makeStyles-root-306 makeStyles-root-317 MuiGrid-container"
                          data-testid="row-5e68b9e6dd814db72fcda7af"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-318 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                ncdai24055@gmail.com
                              </p>
                            </div>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                          >
                            <div
                              class="MuiBox-root MuiBox-root-319 makeStyles-columns__item-309"
                            >
                              <p
                                class="MuiTypography-root makeStyles-columns__item__typo-310 MuiTypography-body2"
                              >
                                ncdai24055
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="resize-triggers"
                  >
                    <div
                      class="expand-trigger"
                    >
                      <div
                        style="width: 201px; height: 701px;"
                      />
                    </div>
                    <div
                      class="contract-trigger"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
